bundles/org.openhab.binding.entsoe/src/main/java/org/openhab/binding/entsoe/internal/EntsoeHandler.java

```diff
@@ -0,0 +1,274 @@
+    }
+
+    private ZonedDateTime currentCetTime() {
+        return ZonedDateTime.now(cetZoneId);
+    }
+
+    private ZonedDateTime currentCetTimeWholeHours() {
+        return currentCetTime().truncatedTo(ChronoUnit.HOURS);
+    }
+
+    private long getMillisToNextStateUpdate() {
+        Instant now = Instant.now();
+        Instant nextHour = now.truncatedTo(ChronoUnit.HOURS).plus(1, ChronoUnit.HOURS);
+        long millis = Duration.between(now, nextHour).toMillis();
+        try {
+            Instant nextInstant = getActualTimeSerie(true).getInstant();
+            millis = Duration.between(now, nextInstant).toMillis();
+        } catch (EntsoeResponseMapException e) {
+            logger.warn("Using millis to next whole hour");
+        }
+        return millis;
+    }
+
+    private EntsoeTimeSerie getActualTimeSerie() throws EntsoeResponseMapException {
+        return getActualTimeSerie(false);
+    }
+
+    private EntsoeTimeSerie getActualTimeSerie(boolean getNext) throws EntsoeResponseMapException {
+        if (entsoeTimeSeries.isEmpty()) {
+            throw new EntsoeResponseMapException("responseMap is empty");
+        }
+        Duration duration = Duration.parse(config.resolution);
+        for (Map.Entry<Instant, EntsoeTimeSerie> entry : entsoeTimeSeries.entrySet()) {
+            Instant thisEntry = entry.getKey();
+            Instant now = Instant.now();
+            Instant nowPlusNext = now.plus(duration);
+            Instant nowMinusNext = now.minus(duration);
+
+            if (getNext && thisEntry.isAfter(now)) {
+                return entry.getValue();
+            }
+            if (!getNext && thisEntry.isAfter(nowMinusNext) && thisEntry.isBefore(nowPlusNext)) {
+                return entry.getValue();
+            }
+        }
+        throw new EntsoeResponseMapException("Could not get actual ENTSOE TimeSerie");
+    }
+
+    private boolean needToFetchHistoricDays() {
+        return needToFetchHistoricDays(false);
+    }
+
+    private boolean needToFetchHistoricDays(boolean updateHistoricDaysInitially) {
+        boolean needToFetch = false;
+        if (historicDaysInitially < config.historicDays) {
+            logger.debug("Need to fetch historic data. Historicdays was changed to a greater number: {}",
+                    config.historicDays);
+            needToFetch = true;
+        }
+
+        if (updateHistoricDaysInitially && historicDaysInitially != config.historicDays) {
+            historicDaysInitially = config.historicDays;
+        }
+
+        return needToFetch;
+    }
+
+    private void refreshPrices() {
+        if (!isLinked(EntsoeBindingConstants.CHANNEL_SPOT_PRICE)) {
+            logger.debug("Channel {} is not linked, skipping channel update",
+                    EntsoeBindingConstants.CHANNEL_SPOT_PRICE);
+            return;
+        }
+
+        config = getConfigAs(EntsoeConfiguration.class);
+
```

Comment:
It would be preferable to cache the deserialized configuration. Actually you did that already in initialize, so you can simple remove this line.
```suggestion
```